src/main/java/org/zalando/nakadi/controller/SubscriptionController.java

```diff
@@ -143,4 +148,35 @@ private Subscription getExistingSubscription(final SubscriptionBase subscription
                 subscriptionBase.getConsumerGroup());
     }
 
+    private final class EventTypesHolder {
```

Comment:
I don't really see a good reason for this class to exist. Why do you need to collect missing ETs? You can always know the missing ones: just subtract found event types from the set of all event types of subscription.